account_invoice_analytic_line/models/account_analytic_line.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    invoice_discount_id = fields.Many2one(
```

Comment:
Do we really need a separate object for discount? On the invoice this is only a float field and I don`t see any special  functionality or so implemented in the  account.invoice.analytic.line.discount module. But if we have an object one needs to create a Discount record each time one use different number.

Reply:
for our use case, we only have fixed discounts, and we want to control which discounts employees can select